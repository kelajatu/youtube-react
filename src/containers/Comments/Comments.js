import React from "react";
import { CommentsHeader } from "./CommentsHeader/CommentsHeader";
import { Comment } from "./Comment/Comment";
import { AddComment } from "./AddComment/AddComment";

export class Comments extends React.Component {
  render() {
    if (!this.props.comments) {
      return <div />;
    }

    const comments = this.props.comments.map(comment => {
      return <Comment comment={comment} key={comment.id} />;
    });
    return (
      <div>
        <CommentsHeader amoountComments={this.props.amoountComments} />
        <AddComment key="add-comment" />
        {comments}
      </div>
    );
  }
}
