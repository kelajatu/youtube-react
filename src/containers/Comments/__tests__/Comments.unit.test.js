import React from "react";
import { shallow } from "enzyme";
import { Comments } from "../Comments";

describe("Comments", () => {
  it("renders Comments without props", () => {
    const wrapper = shallow(<Comments />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders wihout amountComments", () => {
    const wrapper = shallow(<Comments amountComments={123456} />);
    expect(wrapper).toMatchSnapshot();
  });
});
