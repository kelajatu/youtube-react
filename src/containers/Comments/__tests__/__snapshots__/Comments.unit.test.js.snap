// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders Comments without props 1`] = `
<div>
  <CommentsHeader />
  <AddComment />
  <Comment />
  <Comment />
  <Comment />
</div>
`;

exports[`Comments renders wihout amountComments 1`] = `
<div>
  <CommentsHeader />
  <AddComment />
  <Comment />
  <Comment />
  <Comment />
</div>
`;
